src/node_shadow_realm.h

```diff
@@ -3,11 +3,38 @@
 
 #if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
 
+#include "node_realm.h"
 #include "v8.h"
 
 namespace node {
 namespace shadow_realm {
 
+class ShadowRealm : public Realm {
+ public:
+  static ShadowRealm* New(Environment* env);
+
+  SET_MEMORY_INFO_NAME(ShadowRealm)
+  SET_SELF_SIZE(ShadowRealm)
+  void MemoryInfo(MemoryTracker* tracker) const override;
+
+  v8::Local<v8::Context> context() const override;
+
+#define V(PropertyName, TypeName)                                              \
+  v8::Local<TypeName> PropertyName() const override;                           \
+  void set_##PropertyName(v8::Local<TypeName> value) override;
+  PER_REALM_STRONG_PERSISTENT_VALUES(V)
```

Comment:
I think we can also just implement the accessors for the the non-shadow-realm properties with something like `UNREACHABLE()`  in shadow realms since they are already virtual, and that should avoid the casting problem, but I am fine with doing this in a follow-up.